### PR TITLE
Editor further data flow

### PR DESCRIPTION
**Pull request by kaglowka**
- `Viewer` opening on edit button click
- annotation update with `Editor`

### CODE REVIEW

File: src/components/editor/Editor.tsx [hunk 1]
```diff
@@ -45,7 +45,13 @@ import {AnnotationAPICreateModel} from 'api/annotations';
   },
   dispatch => ({
     hideEditor: () => dispatch(hideEditor()),
-    createAnnotation: (data: AnnotationAPICreateModel) => dispatch(createResource(data)),
+    createAnnotation: (model: AnnotationAPICreateModel) => {
```
Comment (lgodziejewski) [inline src/components/editor/Editor.tsx]: why `model`? Wouldn't `data` be fine? (or `annotation`)

File: src/components/viewer/ViewerItem.tsx [hunk 2]
```diff
@@ -60,23 +63,18 @@ export default class ViewerItem extends React.Component<Partial<IViewerItemProps
       ViewerItem.editControlDisappearTimeout,
     );
   }
+  onEditClick = (e) => {
```
Comment (lgodziejewski) [inline src/components/viewer/ViewerItem.tsx]: btw. in React there's a "custom" that event handlers are called `handleXY`, e.g. `onClick={handleClick}`, etc.

File: src/store/widgets/reducers.ts [hunk 3]
```diff
@@ -1,22 +1,28 @@
 import {
-  EDITOR_NEW_ANNOTATION,
+  EDITOR_ANNOTATION,
   EDITOR_VISIBLE_CHANGE,
-  MENU_WIDGET_CHANGE,
+  MENU_WIDGET_CHANGE, SET_EDITOR_SELECTION_RANGE,
   VIEWER_VISIBLE_CHANGE,
 } from './actions';
-import * as _ from "lodash";
+import * as _ from 'lodash';
```
Comment (lgodziejewski) [inline src/store/widgets/reducers.ts]: As a side note, if just a specific function is wanted (and we really care about final package size), it's better to write import as sth like `import _get from 'lodash/get'`, as then only the given fragment gets imported into our bundle.

If we just do a `import foo from 'lodash'`, then (usually) the whole package is injected into our final bundle, which often is not needed.

File: src/components/editor/Editor.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (kaglowka) [reply src/components/editor/Editor.tsx]: @glowka raised the same issue :) I'm changing it in the next PR (already waiting for review). We agreed it's nice to have a universal name (used for many different *models*). We settled for `instance`, but we hadn't thought of `data` then.

File: src/components/viewer/ViewerItem.tsx (hunk 2 above)
Comment (kaglowka) [reply src/components/viewer/ViewerItem.tsx]: I wasn't sure about that, thanks.

File: src/store/widgets/reducers.ts (hunk 3 above)
Comment (kaglowka) [reply src/store/widgets/reducers.ts]: Good advice :+1:

**Pull request merged**